test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js

```diff
@@ -274,3 +285,26 @@ test
     await t.expect(await terminalPageObject.goBackLinkExists()).ok();
     await t.expect(terminalPageObject.getFormTitle()).eql('Verify with your email');
   });
+
+test
+  .requestHooks(logger, dynamicRefreshShortIntervalMock)('dynamic polling based on refresh interval in /poll', async t => {
+    const challengeEmailPageObject = await setup(t);
+    await t.expect(challengeEmailPageObject.resendEmailView().hasClass('hide')).ok();
+
+    // 2 poll requests in 2 seconds at 1 sec interval
+    await t.wait(2000);
+    await t.expect(logger.count(
+      record => record.response.statusCode === 200 &&
+        record.request.url.match(/poll/)
+    )).eql(2);
+
+    await t.removeRequestHooks(dynamicRefreshShortIntervalMock);
+    await t.addRequestHooks(dynamicRefreshLongIntervalMock);
+
+    // 3 poll requests in 6 seconds at 2 sec interval
+    await t.wait(6000);
```

Comment:
Thanks for making the test much efficient..

One thing. This actually doesn't test dynamicRefresh right? it's always 2sec ?

I might have to add sequence of mocks that have different `refreshIntervals`
```
authenticator-verification-email-polling-long-1 - 1000ms
authenticator-verification-email-polling-long-2 - 2000ms
authenticator-verification-email-polling-long-3 - 3000 ms
```

Reply:
It does test it right now, the sequence is as follow:
```
authenticator-verification-email-polling-short-1 - refreshInterval 1000ms - elapsedTime 1000ms - wait 2000ms
authenticator-verification-email-polling-short-2 - refreshInterval 1000ms - elapsedTime 2000ms
authenticator-verification-email-polling-long-1 - refreshInterval 2000 ms - elapsedTime 4000ms - wait 6000ms
authenticator-verification-email-polling-long-2 - refreshInterval 2000ms - elapsedTime 6000ms
authenticator-verification-email-polling-long-3 - refreshInterval 2000 ms - elapsedTime 8000ms

Total Requests: 5
```

Assertion is done at 2000ms to check that there are 2 requests completed so far. Final assertion at the end for 5 total requests at the end.

Reply:
ah okay make sense. thanks for clarification.